doc/graph/fusion_patterns/softmax_fusion_patterns.md

```diff
@@ -18,38 +18,37 @@ brown nodes are optional.
 1. **SoftMax Operation**: Performs the softmax function for the `src` tensor. See
    the [SoftMax](@ref dev_guide_op_softmax) operation in the Graph API for more
    details.
-2. **F2F Conversion Subgraph**: Converts the output tensor from floating-point to
-   another floating-point. It is constructed by a [TypeCast](@ref dev_guide_op_typecast)
-   operation.
-
-   ![f2f_conversion_subgraph](images/f2f_conversion.png)
-
-3. **Epilogue Subgraph**: Optional and can include the following operations:
+2. **Epilogue Subgraph**: Optional and can include the following operations:
    - Binary and Unary operations: refer to the Note in
      [Fusion Patterns](graph_fusion_patterns.html).
 
-   Combination Rules:
-
    ![epilogue subgraph](images/epilogue_subgraph_general_1.png)
 
    - N=20, 0 to 20 Binary or Unary operations are supported in the epilogue
      subgraph.
+3. **F2F Conversion Subgraph**: Converts the output tensor from floating-point
+   to another floating-point. It is constructed by a [TypeCast](@ref dev_guide_op_typecast)
+   operation.
+
+   ![f2f_conversion_subgraph](images/f2f_conversion.png)
 
-4. **F2Q Conversion Subgraph**: Converts the output
-   tensor from floating-point to quantized data type. It can be one of
-   the following subgraphs. See [TypeCast](@ref dev_guide_op_typecast)
-   and [Quantize](@ref dev_guide_op_quantize) operations in Graph API.
+4. **F2Q Conversion Subgraph**: Converts the output tensor from floating-point
+   to quantized data type. It is constructed by a [Quantize](@ref dev_guide_op_quantize)
+   operations in Graph API.
 
-   ![f2q_conversion_subgraph](images/f2q_conversion_softmax.png)
+   ![f2q_conversion_subgraph](images/f2q_conversion_general.png)
 
+5. If multiple optional subgraphs are present, they must follow the order
+   defined in the pattern structure.
 
 ## Data Types
 
-oneDNN supports the following combinations of data types for src and dst:
+Refer to the document of each operation for the supported data types.
 
-| src           | dst             |
-| :------------ | :----------------- |
-| bf16,f16,f32  | u8,s8,bf16,f16,f32 |
+If any optional subgraph is present, the output data type of `SoftMax` must be
+`f32`. With that, the F2F conversion subgraph converts the output from `f32` to
+`f16` or `bf16` and the F2Q conversion sugraph quantizes the output from `f32`
+to `int8`.
```

Comment:
could you explain more about F2F and F2Q?  According to this description, F2F output is f16 or bf16, f2Q input is f32. In the graph [f2q_conversion_softmax.png](https://github.com/uxlfoundation/oneDNN/pull/4464/files#diff-60f04412d9850a14bbf4fd0eb0c9ddeb7c690ecf38f2cdefe8f14b483b45f9f7), F2F output is input of F2Q, which seems incompatible?

Reply:
I think with the data type constraint, it will not be possible to have both F2F and F2Q coexisting here.